EDOT PHP documentation (#144)
elastic/opentelemetry: docs/_edot-sdks/php/setup/index.md

# Setting up EDOT PHP

This guide shows you how to use the Elastic Distribution of OpenTelemetry PHP (EDOT PHP) to instrument your PHP application and send OpenTelemetry data to an Elastic Observability deployment.

**Already familiar with OpenTelemetry?** It's an explicit goal of this distribution to introduce _no new concepts_ outside those defined by the wider OpenTelemetry community.

**New to OpenTelemetry?** This section will guide you through the _minimal_ configuration options to get EDOT PHP set up in your application. You do _not_ need any existing experience with OpenTelemetry to set up EDOT PHP initially. If you need more control over your configuration after getting set up, you can learn more in the [OpenTelemetry documentation](https://opentelemetry.io/docs/languages/php/).

## Prerequisites

Before you begin, make sure you have a destination for the telemetry data collected by EDOT PHP.
While EDOT PHP can export data to any OpenTelemetry Protocol (OTLP)–compatible endpoint, this guide focuses on using [Elastic Observability](https://www.elastic.co/observability) as the backend.
You can either use an existing Elastic Cloud deployment or create a new one.

> To quickly get up and running, follow the [Elastic OpenTelemetry Quickstart guide](https://elastic.github.io/opentelemetry/quickstart/), which walks you through:

Reviewer: ```suggestion
To quickly get up and running, follow the [Elastic OpenTelemetry Quickstart guide](../../quickstart/index), which walks you through:
```

let's use relative links everywhere!

Author: @AlexanderWert  This doesn't work because quickstart is out of fs for this repo. Link-check fails. Should I revert it or you're sure that it will work?

Reviewer: sorry, seems there was one more level of `../` missing in my suggestion, fixed it.

Author: Ah, I was search for quickstart in tree but somehow missed it. Thanks.

- Creating a free Elastic Cloud deployment
- Configuring your OpenTelemetry agent
- Exploring traces and metrics in Kibana

## Install

> ### Prerequisites

Reviewer: we have a prerequisite section above, and one here, it's a bit confusing. Can we consolidate those?

> #### Operating system

Reviewer: That section should go into the `supported-technologies` page, and we can just link it from here

- **Linux**
  - Architectures: **x86_64** and **ARM64**
  - **glibc-based systems**: Packages available as **DEB** and **RPM**
  - **musl libc-based systems (Alpine Linux)**: Packages available as **APK**

#### PHP

Supported PHP versions are 8.1-8.4.
You can find more details in [supported technologies](../supported-technologies.md) doc.

> ### Other limitations

Reviewer: I don't think it's required to link to limitations explicitly here if we will have a limitations page

Author: We're including this link primarily as a precaution, so that the operator installing EDOT PHP can verify that the PHP configuration does not conflict with any known limitations.
This helps prevent unnecessary support cases later on...

See [limitations](./limitations.md) about other limitations of EDOT PHP.

> ### Download and install packages

Reviewer: let's use enumeration syntax here (that makes it easier to read).
So, making it consistent with how it's done in the [Quickstart](https://github.com/elastic/opentelemetry/blob/main/docs/quickstart/self-managed/hosts_vms.md#quickstart)

parent: EDOT PHP
---

<!-- TODO:
- where to download
- explicit description of basic setup here (even if it overlaps with upstream docs)
- link to upstream docs for more advanced setup use cases -->

# Setting up EDOT PHP

This guide shows you how to use the Elastic Distribution of OpenTelemetry PHP (EDOT PHP) to instrument your PHP application and send OpenTelemetry data to an Elastic Observability deployment.

**Already familiar with OpenTelemetry?** It's an explicit goal of this distribution to introduce _no new concepts_ outside those defined by the wider OpenTelemetry community.

**New to OpenTelemetry?** This section will guide you through the _minimal_ configuration options to get EDOT PHP set up in your application. You do _not_ need any existing experience with OpenTelemetry to set up EDOT PHP initially. If you need more control over your configuration after getting set up, you can learn more in the [OpenTelemetry documentation](https://opentelemetry.io/docs/languages/php/).

## Prerequisites

> Before getting started, you'll need somewhere to send the gathered OpenTelemetry data so it can be viewed and analyzed. EDOT PHP supports sending data to any OpenTelemetry protocol (OTLP) endpoint, but this guide assumes you are sending data to an [Elastic Observability](https://www.elastic.co/observability) cloud deployment. You can use an existing one or set up a new one.

Reviewer 2: ```suggestion
Before getting started, you'll need to send the gathered OpenTelemetry data somewhere so it can be viewed and analyzed. EDOT PHP supports sending data to any OpenTelemetry protocol (OTLP) endpoint, but this guide assumes you are sending data to an [Elastic Observability](https://www.elastic.co/observability) cloud deployment. You can use an existing one or set up a new one.
```

<details>
<summary><strong>Expand for setup instructions</strong></summary>

> To create your first Elastic Observability deployment:

Reviewer 2: Can we link the [Quickstart Guide](https://elastic.github.io/opentelemetry/quickstart/) instead of writing the instructions to setup a deployment?

#### Install RPM package (RHEL/CentOS, Fedora)

    rpm -ivh <package-file>.rpm

#### Install DEB package (Debian, Ubuntu 18+)

    dpkg -i <package-file>.deb

#### Install APK package (Alpine)

    apk add --allow-untrusted <package-file>.apk

<!-- Start-to-finish operation -->
## Send data to Elastic

After installing EDOT PHP, configure and initialize it to start sending data to Elastic.

### Configure EDOT PHP

To configure EDOT PHP, at a minimum you'll need your Elastic Observability cloud deployment's OTLP endpoint and
authorization data to set a few `OTLP_*` environment variables that will be available when running EDOT PHP:

* `OTEL_EXPORTER_OTLP_ENDPOINT`: The full URL of the endpoint where data will be sent.
* `OTEL_EXPORTER_OTLP_HEADERS`: A comma-separated list of `key=value` pairs that will
be added to the headers of every request. This is typically used for authentication information.

<!--
These are the instructions used in other distro docs, but in the README in this repo
it looks like you might be recommending using an API key rather than using the secret
token method used in the setup guides in Kibana.
-->

> You can find the values of the endpoint and header variables in Kibana's APM tutorial. In Kibana:

Reviewer 2: In the [Quickstart Guide](https://elastic.github.io/opentelemetry/quickstart/) we link to the Elastic docs describing how to create an API. Does it make sense to reuse the Quickstart Guide here as well?

# Setting up EDOT PHP

This guide shows you how to use the Elastic Distribution of OpenTelemetry PHP (EDOT PHP) to instrument your PHP application and send OpenTelemetry data to an Elastic Observability deployment.

**Already familiar with OpenTelemetry?** It's an explicit goal of this distribution to introduce _no new concepts_ outside those defined by the wider OpenTelemetry community.

**New to OpenTelemetry?** This section will guide you through the _minimal_ configuration options to get EDOT PHP set up in your application. You do _not_ need any existing experience with OpenTelemetry to set up EDOT PHP initially. If you need more control over your configuration after getting set up, you can learn more in the [OpenTelemetry documentation](https://opentelemetry.io/docs/languages/php/).

## 1. Prerequisites

Before you begin, make sure you have a destination for the telemetry data collected by EDOT PHP.
While EDOT PHP can export data to any OpenTelemetry Protocol (OTLP)–compatible endpoint, this guide focuses on using [Elastic Observability](https://www.elastic.co/observability) as the backend.
You can either use an existing Elastic Cloud deployment or create a new one.

> To quickly get up and running, follow the [Elastic OpenTelemetry Quickstart guide](../../quickstart/index), which walks you through:

Reviewer: ```suggestion
To quickly get up and running, follow the [Elastic OpenTelemetry Quickstart guide](../../../quickstart/index), which walks you through:
```
